quickwit-metastore/src/metastore.rs

```diff
@@ -130,6 +132,28 @@ impl Default for SplitState {
     }
 }
 
+impl FromPrimitive for SplitState {
+    fn from_i64(num_split_state: i64) -> Option<SplitState> {
+        match num_split_state {
+            0 => Some(SplitState::New),
+            1 => Some(SplitState::Staged),
+            2 => Some(SplitState::Published),
+            3 => Some(SplitState::ScheduledForDeletion),
+            _ => None,
+        }
```

Comment:
This is useless... 
- From<u64> should be sufficient
- if From<i64> was a good thing, it could still call from_u64.
- anyway let's use strings for enum.